fatal error: pyconfig.h: No such file or directory when cross compiling

I'm seeing this build failure starting with cryptography version `40.0.0`(`39.0.2` builds fine) when cross compiling with [buildroot](https://buildroot.org/).

```rust
     Running `rustc --crate-name build_script_build --edition=2018 /home/buildroot/buildroot/output/build/python-cryptography-41.0.1/VENDOR/pyo3/build.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --diagnostic-width=209 --crate-type bin --emit=dep-info,link -C embed-bitcode=no -C debug-assertions=off -C overflow-checks=on --cfg 'feature="abi3"' --cfg 'feature="abi3-py310"' --cfg 'feature="abi3-py311"' --cfg 'feature="abi3-py37"' --cfg 'feature="abi3-py38"' --cfg 'feature="abi3-py39"' --cfg 'feature="default"' --cfg 'feature="extension-module"' --cfg 'feature="indoc"' --cfg 'feature="macros"' --cfg 'feature="pyo3-macros"' --cfg 'feature="unindent"' -C metadata=11bdde305c3ecfe9 -C extra-filename=-11bdde305c3ecfe9 --out-dir /home/buildroot/buildroot/output/build/python-cryptography-41.0.1/src/rust/target/release/build/pyo3-11bdde305c3ecfe9 -L dependency=/home/buildroot/buildroot/output/build/python-cryptography-41.0.1/src/rust/target/release/deps --extern pyo3_build_config=/home/buildroot/buildroot/output/build/python-cryptography-41.0.1/src/rust/target/release/deps/libpyo3_build_config-36a51f7ec3cba2f9.rlib --cap-lints allow -C link-args=-L/home/buildroot/buildroot/output/per-package/python-cryptography/host/lib -C link-args=-Wl,-rpath,/home/buildroot/buildroot/output/per-package/python-cryptography/host/lib`
The following warnings were emitted during compilation:

warning: In file included from /home/buildroot/buildroot/output/build/python-cryptography-41.0.1/src/rust/target/aarch64-unknown-linux-gnu/release/build/cryptography-cffi-4ab5309155355ef6/out/_openssl.c:57:
warning: /home/buildroot/buildroot/output/build/python3-3.11.4/Include/Python.h:12:10: fatal error: pyconfig.h: No such file or directory
warning:    12 | #include "pyconfig.h"
warning:       |          ^~~~~~~~~~~~
warning: compilation terminated.

error: failed to run custom build command for `cryptography-cffi v0.1.0 (/home/buildroot/buildroot/output/build/python-cryptography-41.0.1/src/rust/cryptography-cffi)`

Caused by:
  process didn't exit successfully: `/home/buildroot/buildroot/output/build/python-cryptography-41.0.1/src/rust/target/release/build/cryptography-cffi-0f846530999218fc/build-script-build` (exit status: 1)
  --- stdout
  cargo:rerun-if-env-changed=PYO3_PYTHON
  cargo:rerun-if-changed=../../_cffi_src/
  cargo:rustc-cfg=python_implementation="CPython"
  TARGET = Some("aarch64-unknown-linux-gnu")
  OPT_LEVEL = Some("3")
  HOST = Some("x86_64-unknown-linux-gnu")
  cargo:rerun-if-env-changed=CC_aarch64-unknown-linux-gnu
  CC_aarch64-unknown-linux-gnu = None
  cargo:rerun-if-env-changed=CC_aarch64_unknown_linux_gnu
  CC_aarch64_unknown_linux_gnu = None
  cargo:rerun-if-env-changed=TARGET_CC
  TARGET_CC = None
  cargo:rerun-if-env-changed=CC
  CC = Some("/home/buildroot/buildroot/output/per-package/python-cryptography/host/bin/aarch64-linux-gcc")
  cargo:rerun-if-env-changed=CFLAGS_aarch64-unknown-linux-gnu
  CFLAGS_aarch64-unknown-linux-gnu = None
  cargo:rerun-if-env-changed=CFLAGS_aarch64_unknown_linux_gnu
  CFLAGS_aarch64_unknown_linux_gnu = None
  cargo:rerun-if-env-changed=TARGET_CFLAGS
  TARGET_CFLAGS = None
  cargo:rerun-if-env-changed=CFLAGS
  CFLAGS = Some("-D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64  -Os -g0 -D_FORTIFY_SOURCE=1")
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some("false")
  CARGO_CFG_TARGET_FEATURE = Some("neon")
  cargo:rerun-if-env-changed=CC_aarch64-unknown-linux-gnu
  CC_aarch64-unknown-linux-gnu = None
  cargo:rerun-if-env-changed=CC_aarch64_unknown_linux_gnu
  CC_aarch64_unknown_linux_gnu = None
  cargo:rerun-if-env-changed=TARGET_CC
  TARGET_CC = None
  cargo:rerun-if-env-changed=CC
  CC = Some("/home/buildroot/buildroot/output/per-package/python-cryptography/host/bin/aarch64-linux-gcc")
  cargo:rerun-if-env-changed=CFLAGS_aarch64-unknown-linux-gnu
  CFLAGS_aarch64-unknown-linux-gnu = None
  cargo:rerun-if-env-changed=CFLAGS_aarch64_unknown_linux_gnu
  CFLAGS_aarch64_unknown_linux_gnu = None
  cargo:rerun-if-env-changed=TARGET_CFLAGS
  TARGET_CFLAGS = None
  cargo:rerun-if-env-changed=CFLAGS
  CFLAGS = Some("-D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64  -Os -g0 -D_FORTIFY_SOURCE=1")
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  CARGO_CFG_TARGET_FEATURE = Some("neon")
  cargo:rerun-if-env-changed=CC_aarch64-unknown-linux-gnu
  CC_aarch64-unknown-linux-gnu = None
  cargo:rerun-if-env-changed=CC_aarch64_unknown_linux_gnu
  CC_aarch64_unknown_linux_gnu = None
  cargo:rerun-if-env-changed=TARGET_CC
  TARGET_CC = None
  cargo:rerun-if-env-changed=CC
  CC = Some("/home/buildroot/buildroot/output/per-package/python-cryptography/host/bin/aarch64-linux-gcc")
  cargo:rerun-if-env-changed=CFLAGS_aarch64-unknown-linux-gnu
  CFLAGS_aarch64-unknown-linux-gnu = None
  cargo:rerun-if-env-changed=CFLAGS_aarch64_unknown_linux_gnu
  CFLAGS_aarch64_unknown_linux_gnu = None
  cargo:rerun-if-env-changed=TARGET_CFLAGS
  TARGET_CFLAGS = None
  cargo:rerun-if-env-changed=CFLAGS
  CFLAGS = Some("-D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64  -Os -g0 -D_FORTIFY_SOURCE=1")
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  CARGO_CFG_TARGET_FEATURE = Some("neon")
  cargo:rerun-if-env-changed=CC_aarch64-unknown-linux-gnu
  CC_aarch64-unknown-linux-gnu = None
  cargo:rerun-if-env-changed=CC_aarch64_unknown_linux_gnu
  CC_aarch64_unknown_linux_gnu = None
  cargo:rerun-if-env-changed=TARGET_CC
  TARGET_CC = None
  cargo:rerun-if-env-changed=CC
  CC = Some("/home/buildroot/buildroot/output/per-package/python-cryptography/host/bin/aarch64-linux-gcc")
  cargo:rerun-if-env-changed=CFLAGS_aarch64-unknown-linux-gnu
  CFLAGS_aarch64-unknown-linux-gnu = None
  cargo:rerun-if-env-changed=CFLAGS_aarch64_unknown_linux_gnu
  CFLAGS_aarch64_unknown_linux_gnu = None
  cargo:rerun-if-env-changed=TARGET_CFLAGS
  TARGET_CFLAGS = None
  cargo:rerun-if-env-changed=CFLAGS
  CFLAGS = Some("-D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64  -Os -g0 -D_FORTIFY_SOURCE=1")
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  CARGO_CFG_TARGET_FEATURE = Some("neon")
  running: "/home/buildroot/buildroot/output/per-package/python-cryptography/host/bin/aarch64-linux-gcc" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-D_LARGEFILE_SOURCE" "-D_LARGEFILE64_SOURCE" "-D_FILE_OFFSET_BITS=64" "-Os" "-g0" "-D_FORTIFY_SOURCE=1" "-I" "/home/buildroot/buildroot/output/build/python3-3.11.4/Include" "-I" "/home/buildroot/buildroot/output/per-package/python-cryptography/host/bin/../aarch64-buildroot-linux-gnu/sysroot/usr/include" "-Wconversion" "-Wno-error=sign-conversion" "-Wno-unused-parameter" "-DPy_LIMITED_API=0x030700f0" "-o" "/home/buildroot/buildroot/output/build/python-cryptography-41.0.1/src/rust/target/aarch64-unknown-linux-gnu/release/build/cryptography-cffi-4ab5309155355ef6/out/2f1fe1929bb62df3-_openssl.o" "-c" "/home/buildroot/buildroot/output/build/python-cryptography-41.0.1/src/rust/target/aarch64-unknown-linux-gnu/release/build/cryptography-cffi-4ab5309155355ef6/out/_openssl.c"
  cargo:warning=In file included from /home/buildroot/buildroot/output/build/python-cryptography-41.0.1/src/rust/target/aarch64-unknown-linux-gnu/release/build/cryptography-cffi-4ab5309155355ef6/out/_openssl.c:57:
  cargo:warning=/home/buildroot/buildroot/output/build/python3-3.11.4/Include/Python.h:12:10: fatal error: pyconfig.h: No such file or directory
  cargo:warning=   12 | #include "pyconfig.h"
  cargo:warning=      |          ^~~~~~~~~~~~
  cargo:warning=compilation terminated.
  exit status: 1

  --- stderr


  error occurred: Command "/home/buildroot/buildroot/output/per-package/python-cryptography/host/bin/aarch64-linux-gcc" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-D_LARGEFILE_SOURCE" "-D_LARGEFILE64_SOURCE" "-D_FILE_OFFSET_BITS=64" "-Os" "-g0" "-D_FORTIFY_SOURCE=1" "-I" "/home/buildroot/buildroot/output/build/python3-3.11.4/Include" "-I" "/home/buildroot/buildroot/output/per-package/python-cryptography/host/bin/../aarch64-buildroot-linux-gnu/sysroot/usr/include" "-Wconversion" "-Wno-error=sign-conversion" "-Wno-unused-parameter" "-DPy_LIMITED_API=0x030700f0" "-o" "/home/buildroot/buildroot/output/build/python-cryptography-41.0.1/src/rust/target/aarch64-unknown-linux-gnu/release/build/cryptography-cffi-4ab5309155355ef6/out/2f1fe1929bb62df3-_openssl.o" "-c" "/home/buildroot/buildroot/output/build/python-cryptography-41.0.1/src/rust/target/aarch64-unknown-linux-gnu/release/build/cryptography-cffi-4ab5309155355ef6/out/_openssl.c" with args "aarch64-linux-gcc" did not execute successfully (status code exit status: 1).


warning: build failed, waiting for other jobs to finish...

    =============================DEBUG ASSISTANCE=============================
    If you are seeing a compilation error please try the following steps to
    successfully install cryptography:
    1) Upgrade to the latest pip and try again. This will fix errors for most
       users. See: https://pip.pypa.io/en/stable/installing/#upgrading-pip
    2) Read https://cryptography.io/en/latest/installation/ for specific
       instructions for your platform.
    3) Check our frequently asked questions for more information:
       https://cryptography.io/en/latest/faq/
    4) Ensure you have a recent Rust toolchain installed:
       https://cryptography.io/en/latest/installation/#rust
    
    Python: 3.11.4
    platform: Linux-5.19.0-45-generic-x86_64-with-glibc2.35
    pip: n/a
    setuptools: 67.8.0
    setuptools_rust: 1.6.0
    rustc: 1.70.0 (90c541806 2023-05-31)
    =============================DEBUG ASSISTANCE=============================
    
error: `cargo rustc --lib --message-format=json-render-diagnostics --manifest-path src/rust/Cargo.toml --target aarch64-unknown-linux-gnu --release -v --features pyo3/extension-module --crate-type cdylib --` failed with code 101
```

--- a/src/rust/cryptography-cffi/build.rs
+++ b/src/rust/cryptography-cffi/build.rs
@@ -47,9 +47,14 @@ fn main() {
     )
     .unwrap();
     println!("cargo:rustc-cfg=python_implementation=\"{}\"", python_impl);
-    let python_include = run_python_script(
+    let python_includes = run_python_script(
         &python,
-        "import sysconfig; print(sysconfig.get_path('include'), end='')",
+        "import os; \
+         import setuptools.dist; \
+         import setuptools.command.build_ext; \
+         b = setuptools.command.build_ext.build_ext(setuptools.dist.Distribution()); \
+         b.finalize_options(); \
+         print(os.pathsep.join(b.include_dirs), end='')",
     )
     .unwrap();
     let openssl_include =
@@ -59,11 +64,14 @@ fn main() {
     let mut build = cc::Build::new();
     build
         .file(openssl_c)
-        .include(python_include)
         .include(openssl_include)
         .flag_if_supported("-Wconversion")
         .flag_if_supported("-Wno-error=sign-conversion")
         .flag_if_supported("-Wno-unused-parameter");
+
+    for python_include in env::split_paths(&python_includes) {
+        build.include(python_include);
+    }
 
     // Enable abi3 mode if we're not using PyPy.
     if python_impl != "PyPy" {

